Mathlib/CategoryTheory/Sites/Whiskering.lean

```diff
@@ -157,4 +159,43 @@ lemma sheafCompose_id : sheafCompose_map (F := F) J (𝟙 _) = 𝟙 _ := rfl
 lemma sheafCompose_comp :
     sheafCompose_map J (η ≫ γ) = sheafCompose_map J η ≫ sheafCompose_map J γ := rfl
 
+end Multicospan
+
+section Preserves
+
+variable [PreservesLimitsOfSize.{v₁} F] [PreservesLimitsOfSize.{v₁} G]
+  [PreservesLimitsOfSize.{v₁} H] (J)
+
+/--
+Composing a sheaf with a functor preserving limits of the same size as the hom sets in `C` yields a
+functor between sheaf categories.
+
+Note: the size of the limit `(S.index P).multicospan` that `F` is required to preserve in
+`CategoryTheory.sheafCompose` is in general larger than this.
+-/
+@[simps]
+def sheafCompose' : Sheaf J A ⥤ Sheaf J B where
```

Comment:
We certainly do not want to duplicate this definition. I am not sure what is the best solution, but one way would be to introduce a class:
```lean
class CategoryTheory.Functor.HasSheafCompose (F) (J : GrothendieckTopology C) : Prop where
  isSheaf (P : Cᵒᵖ ⥤ A) (hP : Presheaf.IsSheaf J P) : Presheaf.IsSheaf J (P ⋙ F)
```
Then, one may define `sheafCompose` under the assumption `F.HasSheafCompose J`, and get separate `HasSheafCompose` instances under `[∀ (X : C) (S : J.Cover X) (P : Cᵒᵖ ⥤ A), PreservesLimit (S.index P).multicospan F]` or `PreservesLimitsOfSize.{v₁, max u₁ v₁} F`. Otherwise, it might be possible to keep the `PreservesLimit (S.index P).multicospan F` assumption, and provide more `PreservesLimits` instances for `WalkingMulticospan`?

Reply:
I introduced the class